### DISCUSSION TITLE
Adding roles and permissions to users in a certain role

### DISCUSSION THREAD

**Discussion by bhagya1101** [directus/directus#16962] [Archive]
I have a role named `Non-Admin` and inside that role, I have 5 users.
 I have a list of around 100 data in my collection

I want to give permission to that 5 users each different 20 data from that collection.

How should I perform it ? Any ideas will be appreciated.

**Comment by br41nslug** [comment]
You can set custom filters in the permissions allowing you to decided who can read/write/update/delete what. For example only allowing you to see files you've uploaded.
![image](https://user-images.githubusercontent.com/9389634/210057245-f6077dc4-9eb0-4b28-bb96-25340aadb32a.png)
https://docs.directus.io/configuration/users-roles-permissions/permissions.html#configure-custom-permissions
